### DISCUSSION TITLE
TS 7056 (The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed) while creating a store

### DISCUSSION THREAD

**Discussion by NathanAP** [vuejs/pinia#2580] [Help and Questions] [answered]
Hello! I'm trying to create a store in my project, but I keep getting ts 7056 (The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed) on its declaration. Any idea on what I could do? 

This is part of my code:
![image](https://github.com/vuejs/pinia/assets/12821171/d7dab781-590b-4555-88e1-cfcc6505663a)

I can get rid of the problem if I 

1 - remove the type on my state;
2 - remove all my getters;
3 - remove my custom classes on type.

Not really sure what I should do because I have 5 others stores and all of them are complex enough to be compared to this one...

Edit: I updated the print to the correct one. Twice.

**Comment by gabriel-rodrigues-42** [comment]
Also happening to me any solution?

**Comment by posva** [comment]
Does anybody have a boiled down reproduction of this I can check?

**Reply by NathanAP** [reply]
I'm not home this weekend, but I'll remember to look what I did or if I'm just ignoring the problem.

I think it had something related to my classes being dependent on each other (like a loop), but I'm not sure right now. Let me check it out on Monday!

**Reply by NathanAP** [reply]
I checked my store and right now there is no error. 

I believe I had a problem in my classes (Agent, Conversation and Dialogue) where they were dependent on each other, so there was a loop that TS could be sad about (and later I found a bigger problem that had relation with that). 

I believe that when I fixed that problem both cases were resolved. Anyway, I'm leaving a print here just to show that my problem is resolved, but I'm not sure about the other 14 people who upvoted this post...

![IMG-20251103-WA0008 1](https://github.com/user-attachments/assets/56f7fc5b-3f08-4a2c-a60f-8438cb5fbcc4)

**Reply by posva** [reply, marked as answer]
Well if anybody has this issue, they can open an issue with a boiled down repro
